salt/metalk8s/orchestrate/upgrade/node.sls

```diff
@@ -0,0 +1,61 @@
+{%- set dest_version = pillar.orchestrate.dest_version %}
+{%- set node_name = pillar.orchestrate.node_name %}
+
+{%- set kubeconfig = "/etc/kubernetes/admin.conf" %}
+{%- set context = "kubernetes-admin@kubernetes" %}
+
+{%- if pillar.metalk8s.nodes[node_name]['version'] == dest_version %}
+
+Node {{ node_name }} already in version {{ dest_version }}:
```

Comment:
Shouldn't we simply call this `deploy_node` or something, and always run everything?

Reply:
You mean even for deploy a node we will drain cordon and uncordon ?

Reply:
Yeah, why not? Would basically be a no-op.

Reply:
Hmm `drain` it's not really idempotent so IMO it's maybe something we want to avoid when we can no ?